hal-core/src/addr.rs

```diff
@@ -132,32 +169,28 @@ impl Address for PAddr {
         self.0 as usize
     }
 
-    fn align_up<A: Into<usize>>(self, _align: A) -> Self {
-        unimplemented!("eliza")
+    #[inline]
+    #[cfg(target_arch = "x86_64")]
+    fn from_usize(u: usize) -> Self {
+        const MASK: usize = 0xFFF0_0000_0000_0000;
+        debug_assert!(
+            u & MASK == 0,
+            "x86_64 physical addresses may not have the 12 most significant bits set!"
+        );
+        Self(u & !MASK)
     }
 
-    /// Aligns `self` down to `align`.
-    ///
-    /// The specified alignment must be a power of two.
-    fn align_down<A: Into<usize>>(self, _align: A) -> Self {
-        unimplemented!("eliza")
-    }
-
-    /// Returns `true` if `self` is aligned on the specified alignment.
-    fn is_aligned<A: Into<usize>>(self, _align: A) -> bool {
-        unimplemented!("eliza")
-    }
-
-    fn as_ptr(&self) -> *const () {
-        unimplemented!("eliza")
+    #[cfg(not(target_arch = "x86_64"))]
```

Comment:
platforms that don't have additional logic for what is considered a valid address can just use the un-cfg'd implementation.

Reply:
Shouldn't that implementation at least use `u` as the value, rather than just `0`?

`Self(0)` as the value for all addresses seems like very surprising behaviour.

Reply:
ag, it was supposed to. OOPS!